include/graphblas/nonblocking/io.hpp

```diff
@@ -388,9 +388,9 @@ namespace grb {
 			"vector"
 		);
 
-		// catch empty mask
+		// If the mask is empty: clear the vector
 		if( size( m ) == 0 ) {
-			return set< descr >( x, val, phase );
```

Comment:
This is actually correct: `size( m ) == 0` is code for `NO_MASK`, so the this was correct